react/CreditCard.tsx

```diff
@@ -62,15 +63,29 @@ const initialDoc = {
 }
 
 interface Props {
-  onCardFormCompleted: (cardLastDigits: string) => void
+  onCardFormCompleted: () => void
   onChangePaymentMethod: () => void
+  cardType: CardType
 }
 
 const CreditCard: React.FC<Props> = ({
   onCardFormCompleted,
   onChangePaymentMethod,
+  cardType,
 }) => {
-  const { paymentSystems, setPaymentField, referenceValue } = useOrderPayment()
+  const {
+    cardLastDigits,
+    payment,
+    paymentSystems,
+    setCardLastDigits,
+    setPaymentField,
+    referenceValue,
+  } = useOrderPayment()
+  const {
+    orderForm: {
+      shipping: { selectedAddress },
+    },
+  } = useOrderForm()
```

Comment:
nit: can we use `useOrderShipping` here?

Reply:
@lucasecdb Actually, we should concentrate all the data on OrderManager, and all the other orchestrators should expose only manipulation functions, IMO. 

What do you think?

Reply:
it doesn't "play" well on some cases. like `order-payment` has the `cardLastDigits` which isn't saved to the orderForm but some components need to read it (like the payment summary). I think we could do a mix of both, if the data is in memory read from the corresponding orchestrator, if not read it from the `order-manager`

Reply:
Agreed!

Reply:
The initial idea behind each orchestrator besides `order-manager` is that they would be the "controller" of their respective segment. With that in mind, using `order-shipping` in a payment context is a bit weird. But nothing is set on stone, and we might one day want to discuss this and define better the role of each orchestrator, because I'm afraid this might become an issue later on.